PatientWebApplication/UserMicroserviceApi/Service/AbstractUserService.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using UserMicroserviceApi.Repository;
+using UserMicroserviceApi.Model;
+
+namespace UserMicroserviceApi.Service
+{
+    public abstract class AbstractUserService<T> where T : Entity
+    {
+        public abstract List<T> GetAll();
+
+        public abstract bool New(T entity);
+
+        public abstract bool Update(T entity);
+
+        public abstract T GetByid(int id);
+
+        public abstract void Remove(T entity);
+
+        public bool isCityValid(string city)
+        {
+            return Regex.Match(city, "^([A-Z][a-zA-Z\\s*]+[,][\\s*]*[A-Z][a-zA-Z\\s*]+[\\s*][0-9]+[a-z]*[,][\\s*]*[0-9]+[,][\\s*]*[A-Z][a-zA-Z\\s*]+)$").Success;
+        }
+        public bool isDataValid(string email, string ucin, T user)
+        {
+
+            bool foundWithinDoctors = isFoundInDoctors(email, ucin, user);
+            bool foundWithinSecretaries = isFoundInSecretaries(email, ucin, user);
+            bool foundWithinManagers = isFoundInManagers(email, ucin, user);
+            if (!foundWithinDoctors || !foundWithinSecretaries || !foundWithinManagers) return false;
+            return true;
+        }
+
+        private bool areUCINsEqual(string firstUCIN, string secondUCIN)
+        {
+            if (firstUCIN.Equals(secondUCIN)) return true;
+            return false;
+        }
+
+        private bool areIDsEqual(int firstID, int secondID)
+        {
+            if (firstID == secondID) return true;
+            return false;
+        }
+
+        private bool areEmailsEqual(string firstEmail, string secondEmail)
+        {
+            if (firstEmail.Equals(secondEmail)) return true;
+            return false;
+        }
+
+        public bool isFoundInManagers(string email, string ucin, T user)
+        {
+            string path = bingPathToAppDir(@"JsonFiles\manager.json");
+            ManagerRepository managerRepository = new ManagerRepository(path);
+            List<ManagerUser> listOfManagers = managerRepository.GetAll();
+
+            foreach (ManagerUser manager in listOfManagers)
+            {
+                if (!areIDsEqual(manager.id, user.id) && (areUCINsEqual(manager.uniqueCitizensidentityNumber, ucin) || areEmailsEqual(manager.email, email)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+
+
+        }
+
+        public bool isFoundInSecretaries(string email, string ucin, T user)
+        {
+            string path = bingPathToAppDir(@"JsonFiles\secretary.json");
+            SecretaryRepository secretaryRepository = new SecretaryRepository(path);
```

Comment:
Write this in form:
"SecretaryRepository secretaryRepository = new SecretaryRepository(bingPathToAppDir(@"JsonFiles\secretary.json"););"